src/core/shelf.sol

```diff
@@ -80,54 +91,105 @@ contract Shelf is TitleOwned {
         shelf[loan].tokenId = nft_;
     }
 
-    function file(uint loan, address registry_, uint nft_, uint principal_) public auth {
-        shelf[loan].registry = registry_;
-        shelf[loan].tokenId = nft_;
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    function token(uint loan) public view returns (address registry, uint nft) {
+        return (shelf[loan].registry, shelf[loan].tokenId);
     }
 
-    function file(uint loan, uint principal_) public auth {
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    // --- Shelf: Loan actions ---
+    function issue(address registry, uint token) public returns (uint) {
+        require(NFTLike(registry).ownerOf(token) == msg.sender, "nft-not-owned");
+
+        bytes32 nft = keccak256(abi.encodePacked(registry, token));
+        require(nftlookup[nft] == 0, "nft-in-use");
+
+        uint loan = title.issue(msg.sender);
+        nftlookup[nft] = loan;
+
+        shelf[loan].registry = registry;
+        shelf[loan].tokenId = token;
+
+        return loan;
     }
 
-    // Move the NFT out of the shelf
-    function move(uint loan, address registry_, uint nft_, address to) public owner(loan) {
-        NFTLike(registry_).transferFrom(address(this), to, nft_);
+    function close(uint loan) public owner(loan) {
+        require(pile.debt(loan) == 0, "outstanding-debt");
+        title.close(loan);
```

Comment:
Why are we doing this? It requires an additional `approve` from the loan owner

Reply:
No, title.close is an auth action and doesn't need an `approve`

Reply:
You are calling the `_burn` method of the ERC721 which requires ownership or approve. 
Ownership is still the borrower and not the shelf (because it is the title)

or I am getting something wrong?

Reply:
as discussed with @manuelpolzhofer moving to separate ticket